src/include/miopen/solver/implicitgemm_ck_util.hpp

```diff
@@ -606,6 +643,12 @@ ConvSolution InitInvokerFactoryNCHW(const ExecutionContext& ctx,
                 std::swap(conv_tensors.xDesc, conv_tensors.yDesc);
             }
 
+            HipEventPtr start = nullptr;
+            HipEventPtr stop  = nullptr;
+            if(handle.IsProfilingEnabled())
+            {
+                ProfilingRecordStart(handle, start, stop);
+            }
```

Comment:
[Recommendation] Rework ProfilingRecordStart/ProfilingRecordStop into a class that holds HipEventPtrs and a reference to handle.
- Ctor should check IsProfilingEnabled() and start profiling, like ProfilingRecordStart().
- Dtor should check IsProfilingEnabled() and then stop profiling and write results into handle (just like ProfilingRecordStop()).

So all the benchmarking stuff in invoker could be reduced to construction of an object just before running the kernels and _implicit_ destruction of that object after the last kernel is enqueued.